moshi/src/main/java/com/squareup/moshi/MapJsonAdapter.java

```diff
@@ -32,16 +35,28 @@
         Type type, Set<? extends Annotation> annotations, Moshi moshi) {
       if (!annotations.isEmpty()) return null;
       Class<?> rawType = Types.getRawType(type);
-      if (rawType != Map.class) return null;
+      if (!SUPPORTED_MAPS.contains(rawType)) return null;
```

Comment:
It's ether this or `if (!(rawType == Map.class || rawType == EnumMap.class))`. But this saves some effort if other map types should be supported in the future.